src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -52,342 +56,3008 @@ const ActorInit Boss_Ganon2_InitVars = {
+            break;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_8090026C.s")
+void func_80902348(BossGanon2* this, GlobalContext* globalCtx) {
+    Player* player;
+    f32 temp_f2;
+    f32 temp_f12;
+    s16 i;
+    s16 j;
+    s16 phi_v0_2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_809002CC.s")
+    if (this->noHitTime == 0) {
+        for (i = 0; i < ARRAY_COUNT(this->unk_864); i++) {
+            if (this->unk_444.elements[i].info.bumperFlags & 2) {
+                this->unk_444.elements[i].info.bumperFlags &= ~2;
+            } else if (this->unk_444.elements[i].info.toucherFlags & 2) {
+                this->unk_444.elements[i].info.toucherFlags &= ~2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900344.s")
+                if (this->unk_312 == 1) {
+                    phi_v0_2 = 0x1800;
+                } else {
+                    phi_v0_2 = 0;
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900580.s")
+                func_8002F6D4(globalCtx, &this->actor, 15.0f, this->actor.yawTowardsPlayer + phi_v0_2, 2.0f, 0);
+                sZelda->unk_3C8 = 8;
+                this->noHitTime = 10;
+                break;
+            }
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900650.s")
+    if (this->unk_324 > 0.0f) {
+        player = PLAYER;
+        temp_f2 = -200.0f - player->actor.world.pos.x;
+        temp_f12 = -200.0f - player->actor.world.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900818.s")
+        if (sqrtf(SQ(temp_f2) + SQ(temp_f12)) > 784.0f) {
+            for (j = 0; j < ARRAY_COUNT(player->flameTimers); j++) {
+                player->flameTimers[j] = Rand_S16Offset(0, 200);
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80900890.s")
+            player->isBurning = true;
+            func_8002F6D4(globalCtx, &this->actor, 10.0f, Math_Atan2S(temp_f12, temp_f2), 0.0f, 0x10);
+            sZelda->unk_3C8 = 8;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80901020.s")
+void func_80902524(BossGanon2* this, GlobalContext* globalCtx) {
+    s8 temp_v0_4;
+    ColliderInfo* acHitInfo;
+    s16 i;
+    u8 phi_v1_2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_8090109C.s")
+    osSyncPrintf("this->no_hit_time %d\n", this->noHitTime);
+    if (this->noHitTime != 0 || ((this->unk_334 == 0) && (this->actionFunc == func_80900890))) {
+        for (i = 0; i < ARRAY_COUNT(this->unk_464); i++) {
+            this->unk_424.elements[i].info.bumperFlags &= ~2;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_8090120C.s")
+    osSyncPrintf("this->look_on %d\n", this->lookOn);
+    if (this->lookOn) {
```

Comment:
I think i would personally leave these og names as comments in the struct and leave them unnamed. At least in the case of `noHitTime`, there will probably be a different name like `invulnerableTimer` or whatever, something less japanese-ey :kappa:

if you decide to commend them in the struct, you can leave them in quotes right beside the variable name. we do this in plenty of other places

Reply:
just saw that this was done because of a review comment above. if i saw it earlier i would have commented lol
@louist103 in the future id avoid suggesting to use og names. It should just be named when a proper name is thought of through documentation